Add Docker Desktop 4.24.1 release notes (#18352)
docker/docs: content/desktop/release-notes.md

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/releases.md).

# 4.24.1

{{< release-date date="2023-10-04" >}}

{{< desktop-install win=true version="4.24.1" build_path="/" >}}

### Bug fixes and enhancements

#### For Windows

> - Fixed a bug on Windows where the Dashboard wouldn't show container logs correctly. Fixes [docker/for-win#13714](https://github.com/docker/for-win/issues/13714).

Reviewer: ```suggestion
- Fixed a bug on Docker Desktop for Windows where the Docker Dashboard wouldn't display container logs correctly. Fixes [docker/for-win#13714](https://github.com/docker/for-win/issues/13714).
```

- Fixed a bug that caused Docker Desktop fail to start on Windows hosts. Fixes [docker/for-win#13662](https://github.com/docker/for-win/issues/13662).
- Modified the Docker Desktop resource saver feature to skip reducing kernel memory on WSL when no containers are running, as this was causing timeouts in some cases. Instead, users are encouraged to enable "autoMemoryReclaim" on WSL directly via the .wslconfig file (available since WSL 1.3.10).

> ### Known Issues

Reviewer: ```suggestion
### Known issues
```

#### For Mac

> - Creating a container with the port 53 fails with the error address `already in use`. As a workaround, deactivate network acceleration by adding `"kernelForUDP": false`, in the settings file located at `~/Library/Group Containers/group.com.docker/settings.json`.

Reviewer: ```suggestion
- Creating a container with the port 53 fails with the error address `already in use`. As a workaround, deactivate network acceleration by adding `"kernelForUDP": false`, in the `settings.json` file located at `~/Library/Group Containers/group.com.docker/settings.json`.
```
